src/jit/lsraxarch.cpp

```diff
@@ -672,66 +689,72 @@ void LinearScan::TreeNodeInfoInit(GenTree* tree)
                     // which allows its second operand to be a contained
                     // immediate wheres xadd instruction requires its
                     // second operand to be in a register.
-                    assert(tree->gtLsraInfo.dstCount == 0);
-
-                    // Give it an artificial type and mark it isLocalDefUse = true.
-                    // This would result in a Def position created but not considered
-                    // consumed by its parent node.
-                    tree->gtType                   = TYP_INT;
-                    tree->gtLsraInfo.isLocalDefUse = true;
+                    assert(info->dstCount == 0);
+
+                    // Give it an artificial type and mark it as an unused value.
+                    // This results in a Def position created but not considered consumed by its parent node.
+                    tree->gtType        = TYP_INT;
+                    info->dstCount      = 1;
+                    info->isLocalDefUse = true;
+                    tree->SetUnusedValue();
                 }
                 else
                 {
-                    assert(tree->gtLsraInfo.dstCount != 0);
+                    assert(info->dstCount != 0);
                 }
 
                 delayUseSrc = op1;
             }
-            else if ((op2 != nullptr) &&
-                     (!tree->OperIsCommutative() || (isContainableMemoryOp(op2) && (op2->gtLsraInfo.srcCount == 0))))
+            else if ((op2 != nullptr) && (!tree->OperIsCommutative() || (op2->isContained() && !op2->IsCnsIntOrI())))
             {
                 delayUseSrc = op2;
             }
-            if (delayUseSrc != nullptr)
+            if ((delayUseSrc != nullptr) && SetDelayFree(delayUseSrc))
```

Comment:
It is a bit confusing reading this now that SetDelayFree is returning a bool.
Perhaps it could be renamed.